src/node_wasi.cc

```diff
@@ -74,22 +74,55 @@ using v8::ArrayBuffer;
 using v8::BackingStore;
 using v8::BigInt;
 using v8::Context;
+using v8::Exception;
 using v8::FunctionCallbackInfo;
 using v8::FunctionTemplate;
+using v8::Integer;
+using v8::Isolate;
 using v8::Local;
 using v8::Object;
 using v8::String;
 using v8::Uint32;
 using v8::Value;
 
 
+static Local<Value> WASIException(Isolate* isolate,
+                                  int errorno,
+                                  const char* syscall) {
+  Environment* env = Environment::GetCurrent(isolate);
+  CHECK_NOT_NULL(env);
+  const char* err_name = uvwasi_embedder_err_code_to_string(errorno);
+  Local<String> js_code = OneByteString(isolate, err_name);
+  Local<String> js_syscall = OneByteString(isolate, syscall);
+  Local<String> js_msg = js_code;
+  js_msg =
+      String::Concat(isolate, js_msg, FIXED_ONE_BYTE_STRING(isolate, ", "));
+  js_msg = String::Concat(isolate, js_msg, js_syscall);
+  Local<Object> e =
+    Exception::Error(js_msg)->ToObject(isolate->GetCurrentContext())
+      .ToLocalChecked();
+
+  e->Set(env->context(),
+         env->errno_string(),
+         Integer::New(isolate, errorno)).Check();
+  e->Set(env->context(), env->code_string(), js_code).Check();
+  e->Set(env->context(), env->syscall_string(), js_syscall).Check();
```

Comment:
I’d be a fan of performing proper error handling here (i.e. returning an empty `MaybeLocal<>` if `Set()` returns an empty `Maybe`)